### PR TITLE
[Process] Throw exception if tempnam returns false

### PR DESCRIPTION
**Pull request by pierredup**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15215 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Process/Pipes/WindowsPipes.php [hunk 1]
```diff
@@ -52,8 +52,7 @@ public function __construct($disableOutput, $input)
                 Process::STDERR => tempnam(sys_get_temp_dir(), 'sf_proc_stderr'),
```
Comment (Tobion) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: > prefix
The prefix of the generated temporary filename.
Note: Windows uses only the first three characters of prefix.

So the prefix would be the same for stdout and stderr on windows, which defeats the purpose. I propose to change the prefix to be different, e.g. `out` and `err`

Comment (pierredup) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: I've changed this to use `out_sf_proc` and `err_sf_proc` as the prefix

**Comment by Tobion** [comment]
:+1: 
Status: reviewed

**Comment by Tobion** [comment]
For 2.3 the same would need to be applied in https://github.com/symfony/symfony/blob/2.3/src/Symfony/Component/Process/ProcessPipes.php#L52

**Comment by pierredup** [comment]
Thanks @Tobion, I have created a new PR for 2.3 here: #16111

**Comment by Tobion** [comment]
Thank you @pierredup.

**Pull request merged**